config/.eslintrc.json

```diff
@@ -0,0 +1,150 @@
+{
+    "env": {
+        "browser": true,
+        "es6": true,
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "plugins": [
+        "@typescript-eslint",
+        "@typescript-eslint/tslint"
+    ],
+    "parserOptions": {
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "overrides": [
+        {
+            "files": [
+                "*test.ts",
+                "**/test/**/*.ts"
+            ],
+            "rules": {
+                "no-unused-expressions": "off"
+            }
+        }
+    ],
+    "rules": {
+        "arrow-body-style": [
+            "error",
+            "as-needed",
+            {
+                "requireReturnForObjectLiteral": true
```

Comment:
But why? :disappointed: I always tell people that they can do `() => ({ ... })` in readability reviews.

Reply:
I was confused by the syntax the first time I saw it, but I don't feel strongly about it.

Reply:
After going through Firestore errors, we are not a fan of "arrow-body-style" at all.  We have 566 violations. 
 Some of them would be fine to change (though unclear benefit), but some would be worse.  For example, it would cause all of our tests to change from, e.g.:
```
  it('can set documents', () => {
    return integrationHelpers.withTestDoc(persistence, doc => {
      return doc.firestore
        .batch()
        .set(doc, { foo: 'bar' })
        .commit()
        .then(() => doc.get())
        .then(snapshot => {
          expect(snapshot.exists).to.equal(true);
          expect(snapshot.data()).to.deep.equal({ foo: 'bar' });
        });
    });
  });
```
to:
```
  it('can set documents', () =>
    integrationHelpers.withTestDoc(persistence, doc =>
      doc.firestore
        .batch()
        .set(doc, { foo: 'bar' })
        .commit()
        .then(() => doc.get())
        .then(snapshot => {
          expect(snapshot.exists).to.equal(true);
          expect(snapshot.data()).to.deep.equal({ foo: 'bar' });
        })
    ));
```

We are not in favor of this, since:
1) We think braces can be a useful tool when coding for denoting long blocks of code (even if it's just chaining method calls).
2) [quoting @wilhuff] Any multiline block that currently contains a single return statement can end up with multiple statements easily. Refactoring the extra statements in is error prone. This is why we require braces around if statements

So I'd advocate for removing this rule at the Firebase level.  And if we keep it, we (Firestore) will likely decide to disable it.

Reply:
Replaced this rule with tslint `"arrow-return-shorthand": true` which is currently enabled in firestore.
It enforces arrow function shorthand, but will ignore functions that span multiple lines (this option is not available in the eslint rule `arrow-body-style`).